projects/sdk/src/lib/silo/Transfer.test.ts

```diff
@@ -0,0 +1,114 @@
+import { TokenValue } from "src/TokenValue";
+import { getTestUtils, ACCOUNTS } from "src/utils/TestUtils/provider";
+import { Transfer } from "./Transfer";
+
+const { sdk, account, utils } = getTestUtils();
+
+jest.setTimeout(30000);
+
+describe("Silo Transfer", function () {
+  const transfer = new Transfer(sdk);
+
+  const SUPPORTED_TOKENS = [sdk.tokens.BEAN, sdk.tokens.BEAN_CRV3_LP, sdk.tokens.UNRIPE_BEAN, sdk.tokens.UNRIPE_BEAN_CRV3];
+
+  const testDestination = ACCOUNTS[1][1];
+
+  it("Fails when using a non-whitelisted token", async () => {
+    const t = async () => {
+      const tx = await transfer.transfer(sdk.tokens.ETH, sdk.tokens.BEAN.amount(3000), testDestination);
+    };
+    expect(t).rejects.toThrow("Transfer error; token ETH is not a whitelisted asset");
+  });
+
+  SUPPORTED_TOKENS.forEach((siloToken) => {
+    describe(`Transfer ${siloToken.symbol}`, () => {
+      describe("Transfer sourced from single crate", () => {
+        beforeAll(async () => {
+          await utils.resetFork();
+
+          // make a deposit
+          await siloToken.approveBeanstalk(TokenValue.MAX_UINT256);
+          // TODO: We can could a new util method to set only the applicable balance
+          await utils.setBEANBalance(account, sdk.tokens.BEAN.amount(2000));
+          await utils.setBEAN3CRVBalance(account, sdk.tokens.BEAN_CRV3_LP.amount(2000));
+          await utils.seturBEANBalance(account, sdk.tokens.UNRIPE_BEAN.amount(2000));
+          await utils.seturBEAN3CRVBalance(account, sdk.tokens.UNRIPE_BEAN_CRV3.amount(2000));
```

Comment:
Why set the balance for all these tokens in the loop? I would expect setting only the current siloToken's balance. The pattern is a little funky but it works. Can prob be improved with a method that accepts the symbol directly. (maybe this is what you meant in my other question)
```
util[`set${siloToken.symbol}Balance`](...)

```

Reply:
I didn't think it warranted extra implementation to do only the applicable token, considering the test suite still runs pretty quickly, but if you disagree I'll take one of the other approaches I mentioned.

Reply:
https://github.com/BeanstalkFarms/Beanstalk/pull/307/commits/c1543e9c8bde1cfe5f450c29db17cb0d4049a5b8